src/MultiFormats/Registry/MultiBaseAlgorithm .cs

```diff
@@ -0,0 +1,224 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+
+namespace MultiFormats.Registry
+{
+    /// <summary>
+    ///   Metadata and implemetations of an IPFS multi-base algorithm.
+    /// </summary>
+    /// <remarks>
+    ///   IPFS assigns a unique <see cref="Name"/> and <see cref="Code"/> to multi-base algorithm.
+    ///   See <see href="https://github.com/multiformats/multibase/blob/master/multibase.csv"/> for
+    ///   the currently defined multi-base algorithms.
+    ///   <para>
+    ///   These algorithms are supported: base58btc, base58flickr, base64,
+    ///   base64pad, base64url, base16, base32, base32z, base32pad, base32hex
+    ///   and base32hexpad.
+    ///   </para>
+    /// </remarks>
+    public class MultiBaseAlgorithm
+    {
+        internal static Dictionary<string, MultiBaseAlgorithm> Names = new Dictionary<string, MultiBaseAlgorithm>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change the visibility of 'Names' or make it 'const' or 'readonly'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)